### ISSUE TITLE
Expose task-specific pre-sbatch commands

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1565]
In SLURM, one may have to run some command _before_ the `sbatch` command (e.g. `module load gpu`, see https://docs.s3it.uzh.ch/cluster/resources).

This doesn't fit within the pattern `extra_lines` we used in other cases (see `"extra_lines": ["#SBATCH --gres=some-requirement"]` below).

```JSON
{
    "default_slurm_config": {
        "partition": "something",
        "cpus_per_task": 1
    },
    "gpu_slurm_config": {
        "partition": "something-else",
        "extra_lines": ["#SBATCH --gres=some-requirement"]
    },
    "batching_config": {
        "target_cpus_per_job": 1,
        "max_cpus_per_job": 1,
        "target_mem_per_job": 200,
        "max_mem_per_job": 500,
        "target_num_jobs": 2,
        "max_num_jobs": 4
    }
}
```

**Comment by tcompa** [comment]
This feature is in-principle implemented as of #1678, but it turns out that it still doesn't work as expected and it it is probably not needed -- see below. Since the situation is unclear (a step is mentioned in the docs but it seems unnecessary), we are currently keeping the `pre_submission_commands` configuration attribute.


## Loading the `module` command in a SSH session
```console
# This works

user@local:~$ ssh clusterloginnode
...

user@clusterloginnode:~$ module list
No modules loaded
user@clusterloginnode:~$ logout

--------------

# This fails

user@local:~$ ssh clusterloginnode  module list
bash: module: command not found
```

## Is it actually needed?
The documentation (https://docs.s3it.uzh.ch/cluster/python_gpu_example/#creating-an-environment-for-tensorflow-on-a-gpu-node) suggests that you need a preliminary `module load gpu`, but it seems that the `gpus` SLURM option is sufficient:

```console
user@loginnode:~$ module list
No modules loaded

user@loginnode:~$ srun --pty -n 1 -c 2 --time=01:00:00 --gpus=1 --mem=8G bash -l

user@computenode:~$ nvidia-smi

[..output of nvidia-smi, listing details of a GPU...]

user@computenode:~$ logout

user@loginnode:~$ module list
No modules loaded
```

For the record, if we remove the `--gpus=1` option then we get an interactive session which doesn't have `nvidia-smi` available ("Command 'nvidia-smi' not found, but can be installed with: ..").

**Comment by tcompa** [comment]
Re: the non-needed `module load gpu`

With @mfranzon we ran the example in https://docs.s3it.uzh.ch/cluster/python_gpu_example/#creating-an-environment-for-pytorch-on-a-gpu-node, but skipped the `module load gpu` step.

The final output is still reasonable:
```
(torch) user@node:~$ python -c 'import torch as t; print("is available: ", t.cuda.is_available()); print("device count: ", t.cuda.device_count()); print("current device: ", t.cuda.current_device()); print("cuda device: ", t.cuda.device(0)); print("cuda device name: ", t.cuda.get_device_name(0)); print("cuda version: ", t.version.cuda)'

is available:  True
device count:  1
current device:  0
cuda device:  <torch.cuda.device object at 0x1514a00046b0>
cuda device name:  Tesla V100-SXM2-16GB
cuda version:  12.4
```

---

Additional info: this is the `module show` (redacted) output:
```
$ module show gpu
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
   ....
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
help([[This module extends the default modules, it adds GPU modules and constrains the resources.

When this module is loaded:

- software installed on under the `gpu` path becomes available
  through the `module` command;

- SLURM `sbatch` and `salloc` commands automatically constrain to
  the feature `gpu`.
]])
family("resource")
prepend_path("MODULEPATH","....")
setenv("SLURM_CONSTRAINT","GPU")
setenv("SALLOC_CONSTRAINT","GPU")
setenv("SBATCH_CONSTRAINT","GPU")
```
